syntax = "proto3";
package imuproto;
message AccelData {
     float x = 1;
     float y = 2;
     float z = 3;
     int32 reserved = 4;
}
message GyroData {
     float x = 1;
     float y = 2;
     float z = 3;
     int32 reserved = 4;
}
message RotVecData {
     // 4 floats for a unit quaternion
     float w = 1;
     float x = 2;
     float y = 3;
     float z = 4;

     float reserved = 5;
}

message  ImuPacket{
     uint64 tsNs = 1;
     AccelData accelData = 2;
     GyroData gyroData = 3;
     RotVecData rotVecData = 4;
}
